### PR TITLE
refactor: move image grid into its own directory

### PR DESCRIPTION
**Pull request by luqven**
This PR refactors the image grid into its own directory for easier and more fine-tuned control over the gird.

### CODE REVIEW

File: src/components/Gallery/ImageGrid.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+import Imgix from 'react-imgix';
+
+import './ImageGrid.css';
+
+interface ImageGridProps {
+  images: string[];
+  onClick: (event: any) => void;
+}
+
+export const ImageGrid: FunctionComponent<ImageGridProps> = ({
```
Comment (sherwinski) [inline src/components/Gallery/ImageGrid.tsx:11]: Do we think it's worth being consistent with how each of these components are implemented, ie using only class or functional components across the board? Or is it perfectly fine to mix the two?

File: src/components/Dialog.tsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ import {
 import { DialogExtensionSDK } from 'contentful-ui-extensions-sdk';
 import { AppInstallationParameters } from './ConfigScreen';
 import ImgixAPI, { APIError } from 'imgix-management-js';
-import Gallery from './Gallery';
+import Gallery from './Gallery/Gallery';
```
[the remark this answers is not in the document]
Comment (luqven) [reply src/components/Dialog.tsx:12]: 💯 , `ImageGrid` or `GalleryComponent`. I'd probably go with `ImageGrid`

File: src/components/Gallery/ImageGrid.tsx (hunk 1 above)
Comment (luqven) [reply src/components/Gallery/ImageGrid.tsx:11]: Don't see an issue with mixing the two. IMO class components make more sense for components who _need_ the react life-cycle methods. Functional components make sense for more presentational components.

Happy to refactor as class comp. if preferred, but my 2c is that func. comps. make more sense for strictly presentational comps. like these.

Although this component does have a "side effect" in that it has an `onClick`. But that will be extracted out to a button in a subsequent commit anyway.

Comment (sherwinski) [reply src/components/Gallery/ImageGrid.tsx:11]: > Happy to refactor as class comp. if preferred, but my 2c is that func. comps. make more sense for strictly presentational comps. like these.

Cool, just wanted to check. This explanation makes sense to me 👍

**Review by sherwinski — APPROVED** [review]

**Pull request merged**